addons/web_tour_recorder/static/src/tour_recorder/tour_recorder_dialog.js

```diff
@@ -0,0 +1,143 @@
+/** @odoo-module */
+
+import { patch } from "@web/core/utils/patch";
+import { useService } from "@web/core/utils/hooks";
+import { browser } from "@web/core/browser/browser";
+import { Dropdown } from "@web/core/dropdown/dropdown";
+import { DropdownItem } from "@web/core/dropdown/dropdown_item";
+import ToursDialog from "@web_tour/debug/tour_dialog_component";
+import { useState, useRef } from "@odoo/owl";
+
+/**
+ * @typedef {import("@web_tour/tour_service/tour_service").Tour} Tour
+ */
+
+patch(ToursDialog, {
+    components: { ...ToursDialog.components, Dropdown, DropdownItem },
+});
+
+patch(ToursDialog.prototype, {
+    setup() {
+        super.setup();
+        this.tourRecorderService = useService("tour_recorder_service");
+        this.notification = useService("notification");
+        const tours = this.tourService.getSortedTours();
+        this.onboardingTours = tours.filter((tour) => !tour.test && !tour.custom);
+        this.testingTours = tours.filter((tour) => tour.test && !tour.custom);
+        this.state = useState({
+            customTours: tours.filter((tour) => tour.custom),
+        });
+        this.inputImportTour = useRef("input_import_tour");
+    },
+
+    /**
+     *
+     * @param {Tour} tour
+     */
+    deleteTour(tour) {
+        const tourIndex = this.state.customTours.indexOf(tour);
```

Comment:
The custom tour state should be managed by the service, if you have multiple instance of this component, the state shouldn't be duplicated. All state held by this component should be UI-related only.